src/clib/grackle_chemistry_data.h

```diff
@@ -99,6 +122,19 @@ typedef struct
   double DeuteriumToHydrogenRatio;
   double SolarMetalFractionByMass;
   double local_dust_to_gas_ratio;
+
+  /* metal/dust abundance */
+
+  int     SN0_N;
+  double *SN0_XC , *SN0_XO , *SN0_XMg, *SN0_XAl, *SN0_XSi, *SN0_XS , *SN0_XFe;
+  double *SN0_fC , *SN0_fO , *SN0_fMg, *SN0_fAl, *SN0_fSi, *SN0_fS , *SN0_fFe;
+  double *SN0_fSiM, *SN0_fFeM, *SN0_fMg2SiO4, *SN0_fMgSiO3, *SN0_fFe3O4
+       , *SN0_fAC, *SN0_fSiO2D, *SN0_fMgO, *SN0_fFeS, *SN0_fAl2O3
+       , *SN0_freforg , *SN0_fvolorg , *SN0_fH2Oice;
+  double *SN0_r0SiM, *SN0_r0FeM, *SN0_r0Mg2SiO4, *SN0_r0MgSiO3, *SN0_r0Fe3O4
+       , *SN0_r0AC, *SN0_r0SiO2D, *SN0_r0MgO, *SN0_r0FeS, *SN0_r0Al2O3
+       , *SN0_r0reforg , *SN0_r0volorg , *SN0_r0H2Oice;
+
```

Comment:
Hi! This is a really cool PR!

I was taking a brief skim through this PR to see if I could help with any of the obstacles to merging in the remaining recent upstream changes since I had introduced the new "dynamic-API" functionality (which is probably the biggest obstacle).

I wanted to ask if there was any reason that these new highlighted fields **need** to be part of the ``chemistry_data`` struct. Would it possibly make more sense to add them into the ``chemistry_data_storage`` struct instead?

I bring this up for a few reasons:
- **Most importantly:** the user doesn't seem to actually specify these values. During the call to ``initialize_chemistry_data``, Grackle seems to entirely ignore whatever values the user may have previously assigned.
- **Less importantly:** These fields feel a little "out-of-place" - since most of them have the type ``double*``.
  - In contrast, all of ``chemistry_data``'s other fields can be thought of as scalar-quantities: they all have a type of ``int``, ``double``, or ``string``.
  - In reality, I know that I'm splitting hairs a little here since C stores a ``string`` as an array of characters. However, I'd still argue these new ``double*`` fields are fundamentally different since they are the only fields within ``chemistry_data`` that Grackle explicitly manages the memory of
- **Least importantly:** doing this would be the easiest approach for merging in the changes relating to the dynamic-api functionality.

I just want to emphasize that my last "reason" is actually a pretty bad argument for you to make this change. If you feel strongly that these fields should be a part of ``chemistry_data``, I'd more than happy to help adjust the dynamic API in order to get this PR merged in. (Currently the dynamic-api functionality assumes that all fields within ``chemistry_data`` are an ``int``, ``double``, or ``string``. We would need to tweak it so that it also knows about ``double*`` fields).

Reply:
I agree we move the SN0_* fields to `chemistry_data_storage` struct.